pkg/image/skopeo.go

```diff
@@ -127,11 +119,12 @@ var extractImageLayers = func(dest string) error {
 			layerID = m.Digest
 		}
 		layer := strings.TrimPrefix(layerID, "sha256:")
-		file := fmt.Sprintf("%s%s/%s", dest, dataTmpDir, layer)
-		err := ExtractTar(file, dest)
-		if err == nil {
-			err = cleanWhiteoutFiles(dest)
+		filePath := fmt.Sprintf("%s%s/%s", dest, dataTmpDir, layer)
+
+		if err := util.UnArchiveLocalTar(filePath, dest, "z"); err != nil {
```

Comment:
Are we guaranteed that filePath will be local, I tried something similar, and it kept feeling when we were using a registry as  the part of the file

Reply:
We are using skopeo to download the tar layers into a local temp folder, then extract it into data folder. In which scenario did it fail for you?